src/rpc/core_rpc_server.cpp

```diff
@@ -190,6 +190,27 @@ namespace cryptonote
     return true;
   }
   //------------------------------------------------------------------------------------------------------------------------------
+  bool core_rpc_server::on_get_blocks_by_height(const COMMAND_RPC_GET_BLOCKS_BY_HEIGHT::request& req, COMMAND_RPC_GET_BLOCKS_BY_HEIGHT::response& res)
+  {
+    CHECK_CORE_BUSY();
+    res.status = "Failed";
+    res.blocks.clear();
+    res.blocks.reserve(req.heights.size());
+    for (uint64_t height : req.heights)
+    {
+      block blk = m_core.get_blockchain_storage().get_db().get_block_from_height(height);
```

Comment:
This can throw, so you may be able to put the daemon in some weird state by asking for a non existent height. I think try try/catch/fail there would ensure this cannot happen.